601-spring-data-primitive-types/src/test/java/io/quarkus/qe/spring/data/CommonsHeadersTest.java

```diff
@@ -0,0 +1,92 @@
+package io.quarkus.qe.spring.data;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import io.restassured.http.Headers;
+import io.restassured.response.Response;
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+
+import static io.restassured.RestAssured.when;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+
+@QuarkusTest
+public class CommonsHeadersTest {
+    // QUARKUS-547
+    @Test
+    @DisplayName("prototype scope: x-count header must be equal to request amount")
+    void testRequestAmount(){
+        Set<String> xCount = new HashSet<>();
+
+        for(int index = 0; index < 3; index++){
+            Headers headers =  when().get("/cat/customFindDistinctivePrimitive/2").headers();
+            xCount.addAll(new HashSet<>(headers.getValues("x-count")));
```

Comment:
Why not adding the collection to the set directly? xCount.addAll(headers.getValues...)